packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -806,36 +806,39 @@ export abstract class FluidDataStoreContext
 			}
 
 			this.thresholdOpsCounter.send("ProcessPendingOps", pending.length);
-			this.pending = undefined;
+		} else {
+			assert(this.pending?.length === 0, "no pending ops");
+
+			// Load the handle to the data store's entryPoint to make sure that for a detached data store, the entryPoint
+			// initialization function is called before the data store gets attached and potentially connected to the
+			// delta stream, so it gets a chance to do things while the data store is still "purely local".
+			// This preserves the behavior from before we introduced entryPoints, where the instantiateDataStore method
+			// of data store factories tends to construct the data object (at least kick off an async method that returns
+			// it); that code moved to the entryPoint initialization function, so we want to ensure it still executes
+			// before the data store is attached.
+			await channel.entryPoint.get();
```

Comment:
I am not sure if this is the right thing to do because it will run application / user code. One implication is that any failure here will be perceived as a runtime failure as happens when instantiating existing data stores. Maybe we should wrap this in a try catch block and throw a UsageError or something similar to indicate the failure is because of user code.

Reply:
This flow is only for existing === false, so it does not apply to instantiating existing data stores.
It's wrapped in try/catch (at a caller level) - see earlier comment

Reply:
Yes, I know its for only creation but I believe we should clearly indicate that the data store failed to create because of a user error.